samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c

```diff
@@ -309,11 +341,11 @@ static void gen_onoff_set_unack(struct bt_mesh_model *model,
 				struct bt_mesh_msg_ctx *ctx,
 				struct net_buf_simple *buf)
 {
-	u8_t tid, tmp8;
+	u8_t tid;
 	struct net_buf_simple *msg = model->pub->msg;
 	struct generic_onoff_state *state = model->user_data;
 
-	tmp8 = net_buf_simple_pull_u8(buf);
+	state->tmp = net_buf_simple_pull_u8(buf);
```

Comment:
try to use more descriptive names or at least provide some comment what is the first byte in this message, so that if someone is reading your code knows what you assign to state->tmp without looking into the specification

Reply:
Here tmp is temporary variable only.
If it is greater than 1 then it will not allow to assign any value to actual onoff state variable
It also get access in gen_onoff_set(), based on which it will not publish if received gen_onoff
state is > 1. Hence put it in user's **struct generic_onoff_state**

Reply:
Please rename it to something more meaningful. `tmp` is really only ok for a temporary stack variable which the function uses for multiple different purposes. In this case it seems single-purpose and it's not temporary (it's part of of state struct instead of being a stack variable). It seems to me like it contains the current OnOff state value, so why not call it `val`?

Reply:
Ok. I will do it shortly.